src/test/java/uk/gov/hmcts/reform/blobrouter/services/SasTokenGeneratorServiceTest.java

```diff
@@ -56,20 +65,26 @@ void should_throw_exception_when_requested_service_is_not_configured() {
             .hasMessage("No service configuration found for nonexistingservice");
     }
 
+    @Test
+    void should_throw_exception_when_service_is_disabled() {
+        assertThatThrownBy(() -> tokenGeneratorService.generateSasToken(DISABLED_SERVICE))
+            .isInstanceOf(ServiceConfigNotFoundException.class)
```

Comment:
should it be disable service or something instead of config not found ?

Reply:
I thought it would be best to not differentiate between those two scenarios.

Reply:
Differentiating  would make it clear in my opinion

Reply:
It **is** configured, but it is **disabled**

Reply:
ok, I'll change this

Reply:
done. Added new exception type, used different message